plugins/PrepareStage/PrepareMenu.qml

```diff
@@ -19,53 +19,102 @@ Item
+            RowLayout
+            {
+                id: itemRow
+
+                anchors.centerIn: parent
+
+                width: 0.9 * prepareMenu.width
+                height: parent.height
+                spacing: 0
+
+                Cura.MachineSelector
+                {
+                    id: machineSelection
+                    z: openFileButtonBackground.z - 1
+                    Layout.minimumWidth: UM.Theme.getSize("machine_selector_widget").width
+                    Layout.maximumWidth: UM.Theme.getSize("machine_selector_widget").width
+                    Layout.fillWidth: true
+                    Layout.fillHeight: true
+                }
+
+                // Separator line
+                Rectangle
+                {
+                    height: parent.height
+                    width: UM.Theme.getSize("default_lining").width
+                    color: UM.Theme.getColor("lining")
+                }
+
+                Cura.QuickConfigurationSelector
+                {
+                    Layout.fillHeight: true
+                    Layout.fillWidth: true
+                    Layout.preferredWidth: itemRow.width - machineSelection.width - printSetupSelector.width - 2 * UM.Theme.getSize("default_lining").width
+                }
+
+                // Separator line
+                Rectangle
+                {
+                    height: parent.height
+                    width: UM.Theme.getSize("default_lining").width
+                    color: UM.Theme.getColor("lining")
+                }
+
+                Cura.PrintSetupSelector
+                {
+                    id: printSetupSelector
+
+                    onShowTooltip: prepareMenu.showTooltip(item, location, text)
+                    onHideTooltip: prepareMenu.hideTooltip()
+
+                    Layout.minimumWidth: UM.Theme.getSize("print_setup_widget").width
+                    Layout.maximumWidth: UM.Theme.getSize("print_setup_widget").width
+                    Layout.fillWidth: true
+                    Layout.fillHeight: true
+                }
+            }
         }
 
-        Cura.PrintSetupSelector
+        Rectangle
         {
-            width: UM.Theme.getSize("print_setup_widget").width
-            onShowTooltip: prepareMenu.showTooltip(item, location, text)
-            onHideTooltip: prepareMenu.hideTooltip()
+            id: openFileButtonBackground
+            height: UM.Theme.getSize("stage_menu").height
+            width: UM.Theme.getSize("stage_menu").height
+
+            radius: UM.Theme.getSize("default_radius").width
+            color: UM.Theme.getColor("toolbar_background")
+            Button
+            {
+                id: openFileButton
+                text: catalog.i18nc("@action:button", "Open File")
+                iconSource: UM.Theme.getIcon("load")
+                style: UM.Theme.styles.toolbar_button
+                tooltip: ""
```

Comment:
Superfluous, I think.
```suggestion
```

Reply:
Nope! Otherwise, it gets two tooltips (one from the button, one from the theme).